infra/deploy.sh

```diff
@@ -343,17 +343,22 @@ deployAzureResources () {
         --parameters "enablePrivateEndpoints=$ENABLE_PRIVATE_ENDPOINTS" \
         --parameters "acrName=$CONTAINER_REGISTRY_NAME" \
         --output json)
+    # errors in deployment may not be caught by exitIfCommandFailed function so we also check the output for errors
     exitIfCommandFailed $? "Error deploying Azure resources..."
+    exitIfValueEmpty "$AZURE_DEPLOY_RESULTS" "Error deploying Azure resources..."
```

Comment:
Based on some recent failed deployments, I noticed that when the azure deployment fails, the `exitIfCommandFailed()` function does not catch the error because the assignment of the execution of azure cli gets assigned to an environment variable (regardless of success/failure). By checking if the env variable is empty, we can exit the script early if the deployment was truly a failure.